### PR TITLE
feat(alwaysontop): add dismiss button and event

### PR DESCRIPTION
**Pull request by gimre**

### CODE REVIEW

File: alwaysontop/main.js [hunk 1]
```diff
@@ -64,9 +64,9 @@ function onAlwaysOnTopWindow(
                 titleBarStyle: undefined,
                 frame: false,
                 show: false,
-                webPreferences: {
+                webPreferences: Object.assign({}, options.webPreferences, {
```
Comment (hristoterezov) [inline alwaysontop/main.js]: Why?

File: alwaysontop/constants.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
+    ALWAYSONTOP_DISMISSED: 'dismissed',
```
Comment (hristoterezov) [inline alwaysontop/constants.js]: Do we need this event? Why?

File: alwaysontop/render.js [hunk 3]
```diff
@@ -319,9 +319,17 @@ class AlwaysOnTop extends EventEmitter {
             this._alwaysOnTopWindow.document.body.innerHTML = `
               <div id="react"></div>
               <video autoplay="" id="video" style="transform: none;" muted></video>
+              <div class="dismiss">&times;</div>
               <link rel="stylesheet" href="file://${ cssPath }">
             `;
 
+            this._alwaysOnTopWindow.document.body.addEventListener('click', e => {
```
Comment (hristoterezov) [inline alwaysontop/render.js]: isn't it better to attach the listener on the element instead of the body?

File: alwaysontop/constants.js (hunk 2 above)
Comment (gimre) [reply alwaysontop/constants.js]: to know when the window was dismissed

File: alwaysontop/main.js (hunk 1 above)
Comment (gimre) [reply alwaysontop/main.js]: to keep webPreferences set upstream, not overwrite them

File: alwaysontop/render.js (hunk 3 above)
Comment (gimre) [reply alwaysontop/render.js]: in this case this is more explicit, because we are setting innerHTML

File: alwaysontop/main.js (hunk 1 above)
Comment (hristoterezov) [reply alwaysontop/main.js]: I figured that. My question is more like:
How this relates to the PR? Are you trying yo solve an issue with this? What is the issue? What is the need of this change?

If this is not related to the PR, please open another PR with this change and put some description there what are you trying to solve!

File: alwaysontop/constants.js (hunk 2 above)
Comment (hristoterezov) [reply alwaysontop/constants.js]: How are you consuming this event?

Comment (gimre) [reply alwaysontop/constants.js]: i am consuming it from the requiring module to have logic that would prevent the window being displayed if it is dimissed by the user

File: alwaysontop/main.js (hunk 1 above)
Comment (gimre) [reply alwaysontop/main.js]: it was just something I noticed while trying different approaches - I'll revert this change

File: alwaysontop/render.js (hunk 3 above)
Comment (hristoterezov) [reply alwaysontop/render.js]: ok but performance-wise, isn't it going to be better to put the lister on the element?

Also I'm wondering if alwaysontop.js is better place for this code. WDYT?

Comment (gimre) [reply alwaysontop/render.js]: doesn't matter tbh but found a workaround

**Comment by gimre** [comment]
> Left some comments and questions!
> 
> Also when I tested the 'x' is not aligned very well, it's closer to the bottom of the circle:
> Screenshot 2020-01-29 at 11 05 52 AM

what OS are you using? kind of looks like some font problem somehow

**Comment by gimre** [comment]
seems `display: flex` and text alignment have some quirks when using different fonts - reimplemented without using text

File: alwaysontop/alwaysontop.js [hunk 4]
```diff
@@ -5,7 +5,8 @@ const {
     onload,
     onbeforeunload,
     shouldImplementDrag,
-    getCurrentSize
+    getCurrentSize,
+    dismiss,
```
Comment (hristoterezov) [inline alwaysontop/alwaysontop.js]: redundant `,`

File: alwaysontop/alwaysontop.js [hunk 5]
```diff
@@ -15,6 +16,9 @@ const {
  */
 let initialSize;
 
+
+document.querySelector('.dismiss').addEventListener('click', dismiss);
```
Comment (hristoterezov) [inline alwaysontop/alwaysontop.js]: please add a check for the element

**Pull request merged**